src/app/src/main/kotlin/org/imperial/mrc/hint/exceptions/HintExceptionHandler.kt

```diff
@@ -28,25 +27,27 @@ import java.lang.reflect.UndeclaredThrowableException
 import java.net.BindException
 import java.text.MessageFormat
 import java.util.*
+import javax.servlet.http.HttpServletRequest
 
 @Order(Ordered.HIGHEST_PRECEDENCE)
 @ControllerAdvice
 class HintExceptionHandler(private val errorCodeGenerator: ErrorCodeGenerator,
-                           private val appProperties: AppProperties)
+                           private val appProperties: AppProperties,
+                           private val logger: GenericLogger
+)
 {
-    private val logger = LogFactory.getLog(HintExceptionHandler::class.java)
 
     @ExceptionHandler(NoHandlerFoundException::class)
-    fun handleNoHandlerFoundException(error: Exception, request: WebRequest): Any
+    fun handleNoHandlerFoundException(error: Exception, request: HttpServletRequest): Any
```

Comment:
Because WebRequest does not have all requests methods I need to capture full logs and to also use same request Type for logs since `HttpServletRequest` is used in controller.